client/my-sites/earn/memberships/index.jsx

```diff
@@ -697,10 +705,25 @@ class MembershipsSection extends Component {
 	}
 }
 
+/**
+ * Source is used to add data to the Stripe Connect URL. On a successful
+ * connection, this source is used to redirect the user the appropriate place.
+ */
+const getSource = () => {
+	if ( window.location.hash === ADD_NEWSLETTER_PAYMENT_PLAN_HASH ) {
+		return 'earn-newsletter';
+	}
+	if ( window.location.hash === LAUNCHPAD_HASH ) {
+		return 'launchpad';
+	}
+	return 'calypso';
```

Comment:
Everything is in "calypso" so would hope this to be something more specific too.

We also have starting points from the block editor directly at least on Jetpack sites.

Reply:
I think I agree, but for now I'm just pulling the 'calypso' default from line 725 below. So this PR is just preserving the current default set on this page.